pkg/operator/discover/discover.go

```diff
@@ -211,6 +212,14 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage, securityAcc
 		}
 	}
 
+	podLabels := os.Getenv(discoverDaemonSetPodLabelsEnv)
+	if podLabels != "" {
+		podLabels := k8sutil.ParseStringToLabels(podLabels)
+		// Set the app label even if set by user as otherwise the DaemonSet pod selector breaks
+		podLabels["app"] = discoverDaemonsetName
```

Comment:
Is `app` the only label on the discover pod right now? If not, the others would be lost with this change.

Reply:
Yes, that is the only label.
I have just seen that the discover DaemonSet doesn't have any labels added to it and added that the `app` label is set on it.